modules/tinymce/src/plugins/anchor/main/ts/core/Anchor.ts

```diff
@@ -5,37 +5,88 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
+import { Node, HTMLAnchorElement } from '@ephox/dom-globals';
+import { Arr, Option } from '@ephox/katamari';
 import Editor from 'tinymce/core/api/Editor';
-import { Element } from '@ephox/dom-globals';
+import RangeUtils from 'tinymce/core/api/dom/RangeUtils';
+import * as Settings from '../api/Settings';
 
-const isNamedAnchor = (editor: Editor, node: Element) => node.tagName === 'A' && editor.dom.getAttrib(node, 'href') === '';
+const namedAnchorSelector = 'a:not([href])';
+
+const isAnchor = (elm: Node): elm is HTMLAnchorElement => elm && elm.nodeName.toLowerCase() === 'a';
+const isNamedAnchor = (elm: Node): elm is HTMLAnchorElement => isAnchor(elm) && !elm.getAttribute('href') && (!!elm.getAttribute('id') || !!elm.getAttribute('name'));
```

Comment:
Generally we try to avoid using `!!`, so could this be written differently? If not, then I believe you can move it outside the parentheses.

Reply:
Actually, why is the `!!` even needed here? Don't we just want to check if either has a value?

Reply:
Also this is a very long line, please break it up over 2 lines. eg:
```
const isNamedAnchor = (elm: Node): elm is HTMLAnchorElement =>
  isAnchor(elm) && !elm.getAttribute('href') && (!!elm.getAttribute('id') || !!elm.getAttribute('name'));

```

Reply:
`isNamedAnchor` would return a string rather than a boolean without the `!!`. I have restructured the function so it should be a bit cleaner now.